includes/Core/Remote_Features/Remote_Features_Provider.php

```diff
@@ -104,7 +126,44 @@ protected function on_admin_init() {
 		}
 
 		$this->cron->maybe_schedule_cron();
+		$this->remote_features_sync_fallback();
 		// Sync remote features when credentials change (e.g. during setup).
 		$this->credentials->on_change( array( $this->syncer, 'pull_remote_features' ) );
 	}
+
+	/**
+	 * Fallback for syncing the remote features if CRON did not run.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function remote_features_sync_fallback() {
```

Comment:
The reason E2E is failing with "too many connections" is due to the following:
This method is called on `admin_init`, which invokes a request to `admin-ajax.php`, which also triggers `admin_init` (repeat). If you observe the access log in Docker, you can see `admin-ajax.php` being called sometimes 50x in a row between any other calls and over 500x within a few minutes. I see the transient which is intended to prevent this, but that isn't working as expected. Generally when there is a known possibility of a race condition, this kind of flag should be set before the main action (request in this case) to minimize the time/potential. 

---

As a secondary point, please extract this to a class to better follow the convention established here of having leaner, purpose-specific classes. Ideally, the existing classes shouldn't need to change here at all, with the exception of a few minimal additions in this one.

---

Finally, nothing related to the fallback here should have any effect unless proxy credentials are present (using_proxy does not imply this). The other actions here like scheduling the cron and registering a listener are okay to happen before this because of what they do. This method will always fail if proxy credentials aren't present to start with.